src/commonMain/kotlin/app/softwork/routingcompose/Path.kt

```diff
@@ -0,0 +1,26 @@
+package app.softwork.routingcompose
+
+internal data class Path(val path: String, val parameters: Parameters?) {
+    fun newPath(currentPath: String) = Path(path = path.removePrefix("/$currentPath"), parameters)
+
+    internal companion object {
+        fun from(rawPath: String): Path {
+            require(rawPath.startsWith("/")) { "path must start with a slash: $rawPath" }
```

Comment:
This `rawPath.startsWith("/")` is quite redundant and odd to work with given hash router. I'd suggest instead just allowing both cases and then simply processing with `path.removePrefix("/")`

Reply:
What do you mean? :) The `require` just moved from `Router` to the new class `Path`, but was already used in older versions.
The URL handling is the same as before, and uses the same parsing as React HashRouter: https://v5.reactrouter.com/web/api/HashRouter

Reply:
Yeah, I meant that slash should be optional, regardless of where require is. :D